### DISCUSSION TITLE
Vídeo 05 - Desenvolvimento

### DISCUSSION THREAD

**Discussion by thpoiani** [thpoiani/lab-quarkus#6] [General]
Esse vídeo foi um pouco complicado para mim 😅 
Meu planejamento era usar **[Hibernate Reactive with Panache](https://quarkus.io/guides/hibernate-reactive-panache)** para aproveitar a linguagem reativa do Quarkus, contudo ainda existem alguns problemas de compatibilidade para usar Flyway com conexão reativa. Existe um _workaround_ para essa issue https://github.com/quarkusio/quarkus/issues/10716, mas isso trouxe outros problemas com o Testcontainers. Decidi voltar ao [Hibernate ORM](https://quarkus.io/guides/hibernate-orm) padrão.

Quem nunca sofreu achando que seria uma simples implementação, mas no final levou horas para resolver um problema?
Compartilhe uma história parecida 😄

**Comment by didifive** [comment]
Acho que é quase uma rotina para programador enfrentar essa de tentar implementar uma ideia que parece muito legal, mas esbarrar em problemas que ainda temos que aprofundar para entender melhor e entender até onde podemos ir.
Essa questão de levar horas já passe pelo clássico centralizar div no HTML, passei por problema de uso de beans no spring com apache camel, até entender que eu estava errando criando instancia nova fora do contexto do spring. Estes são os que lembro.
Trabalhar com programação é um eterno troubleshooting

**Comment by mauromortoni** [comment]
Fui contratado por uma empresa particular para atuar na Caixa Econômica Federal, e minha primeira tarefa consistia em solucionar alguns problemas relacionados a HTML em um sistema, o que deveria levar, no máximo, 4 horas. No entanto, levei um mês para concluir essa tarefa.

O atraso foi causado por diversas questões burocráticas, como a criação de permissões para meu equipamento e usuário. Além disso, precisei portar a aplicação do Sun Java System Application Server (SJSAS) 8.2 para o JBoss EAP 6.4, para então corrigir os HTMLs. 

O suporte e gerenciamento do sistema eram realizados em São Paulo, que estava descontinuando esse suporte devido à obsolescência do sistema. Para complicar ainda mais, a transferência do sistema para Brasília ocorreu em um momento em que a equipe local não possuía conhecimento suficiente sobre a aplicação.
